Client/BraveSkus/BraveSkusManager.swift

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+import BraveShared
+import BraveCore
+import BraveVPN
+import os.log
+
+public class BraveSkusManager {
+  private let sku: SkusSkusService
+  private let considerCachedCredentials: Bool
+  
+  public init?(isPrivateMode: Bool, considerCachedCredentials: Bool) {
+    guard let skusService = Skus.SkusServiceFactory.get(privateMode: isPrivateMode) else {
+      assertionFailure("Failed to create SkusService")
+      return nil
+    }
+    
+    self.sku = skusService
+    self.considerCachedCredentials = considerCachedCredentials
+  }
+  
+  public static func refreshSKUCredential(isPrivate: Bool) {
+    guard let _ = Preferences.VPN.skusCredential.value,
+          let domain = Preferences.VPN.skusCredentialDomain.value,
+          let expirationDate = Preferences.VPN.skusCredentialExpirationDate.value else {
+      Logger.module.debug("No skus credentials stored in the app.")
+      return
```

Comment:
Do not refresh credentials if older credential does not exist. The first/initial credential must be grabbed on the account.brave website